crypto/asn1/x_pubkey.c

```diff
@@ -72,6 +72,12 @@
 static int pubkey_cb(int operation, ASN1_VALUE **pval, const ASN1_ITEM *it,
                      void *exarg)
 {
+    if (operation == ASN1_OP_NEW_POST) {
+        X509_PUBKEY *pubkey = (X509_PUBKEY *)*pval;
+        pubkey->lock = CRYPTO_THREAD_lock_new();
+        if (pubkey->lock == NULL)
+            return 0;
+    }
```

Comment:
The lock needs to be allocated somewhere since there's no `X509_PUBKEY_new` function where we can do that. I noticed that the lock didn't get freed though (in the `ASN1_OP_FREE_POST` phase), so I fixed that now.